openapi-cli/src/main/java/io/ballerina/generators/auth/BallerinaAuthConfigGenerator.java

```diff
@@ -0,0 +1,487 @@
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createObjectFieldNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createOptionalFieldAccessExpressionNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createOptionalTypeDescriptorNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createParenthesizedArgList;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createPositionalArgumentNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createRequiredParameterNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createSeparatedNodeList;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createSimpleNameReferenceNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createToken;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createTypedBindingPatternNode;
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createVariableDeclarationNode;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.CHECK_KEYWORD;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.CLOSE_PAREN_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.COMMA_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.DOT_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.EQUAL_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.QUESTION_MARK_TOKEN;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.SEMICOLON_TOKEN;
+import static io.ballerina.generators.GeneratorConstants.API_KEY;
+import static io.ballerina.generators.GeneratorConstants.API_KEY_CONFIG;
+import static io.ballerina.generators.GeneratorConstants.API_KEY_CONFIG_PARAM;
+import static io.ballerina.generators.GeneratorConstants.API_KEY_CONFIG_RECORD_FIELD;
+import static io.ballerina.generators.GeneratorConstants.API_KEY_MAP;
+import static io.ballerina.generators.GeneratorConstants.AUTH_CONFIG_FILED_NAME;
+import static io.ballerina.generators.GeneratorConstants.AuthConfigTypes;
+import static io.ballerina.generators.GeneratorConstants.BASIC;
+import static io.ballerina.generators.GeneratorConstants.BEARER;
+import static io.ballerina.generators.GeneratorConstants.CONFIG_RECORD_ARG;
+import static io.ballerina.generators.GeneratorConstants.CONFIG_RECORD_NAME;
+import static io.ballerina.generators.GeneratorConstants.HTTP;
+import static io.ballerina.generators.GeneratorConstants.OAUTH2;
+import static io.ballerina.generators.GeneratorConstants.SSL_FIELD_NAME;
+import static io.ballerina.generators.GeneratorUtils.escapeIdentifier;
+
+/**
+ * This class is used to generate authentication related nodes of the ballerina connector client syntax tree.
+ */
+public class BallerinaAuthConfigGenerator {
+    private static final List<String> headerApiKeyNameList = new ArrayList<>();
+    private static final List<String> queryApiKeyNameList = new ArrayList<>();
+    private static boolean isAPIKey = false;
```

Comment:
Better to use noun to represent an object state. 
`private static boolean apiKey = false;`

It also improves the readability
```java
if (apiKey) {
    typeName = AbstractNodeFactory.createIdentifierToken(API_KEY_CONFIG);
}
```